### PR TITLE
Fix missing dependencies due to Aggregation through Join pushdown

### PR DESCRIPTION
**Pull request by kasiafi**
Fixes https://github.com/prestosql/presto/issues/6138

### CODE REVIEW

File: presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java [hunk 1]
```diff
@@ -122,6 +122,53 @@ public void testPushesAggregationThroughRightJoin()
...
     }
 
+    @Test
+    public void testPushesAggregationWithMask()
+    {
+        tester().assertThat(new PushAggregationThroughOuterJoin())
+                .on(p -> p.aggregation(ab -> ab
+                        .source(
+                                p.join(
+                                        JoinNode.Type.LEFT,
```
Comment (sopel39) [inline presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java:132]: static import `LEFT` and `INNER`

File: presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java [hunk 2]
```diff
@@ -122,6 +122,53 @@ public void testPushesAggregationThroughRightJoin()
...
+                                        p.values(ImmutableList.of(p.symbol("COL1")), ImmutableList.of(expressions("10"))),
+                                        p.values(p.symbol("COL2"), p.symbol("MASK")),
+                                        ImmutableList.of(new JoinNode.EquiJoinClause(p.symbol("COL1"), p.symbol("COL2"))),
```
Comment (sopel39) [inline presto-main/src/test/java/io/prestosql/sql/planner/iterative/rule/TestPushAggregationThroughOuterJoin.java:135]: static import `EquiJoinClause`

**Review by sopel39 — APPROVED** [review]

**Comment by sopel39** [comment]
BTW: I think this rule can be generalized. Currently there is:
```
    private static boolean groupsOnAllColumns(AggregationNode node, List<Symbol> columns)
    {
        return node.getGroupingSetCount() == 1 && new HashSet<>(node.getGroupingKeys()).equals(new HashSet<>(columns));
    }
```
but it seems that grouping keys must just contain all symbols from outer side

**Review by martint — APPROVED** [review]

**Comment by kasiafi** [comment]
>BTW: I think this rule can be generalized.

I applied the other comments. This could be looked into as a follow-up.

**Pull request merged**